sql-project-2/sql-project-2_solution.sql

```diff
@@ -0,0 +1,98 @@
+#1 Create a temp table to store the row quantity of each table in “loandb” and the temp table includes 2 columns,
+# one is “table_name” and the other is “row_quantity.” Show the table in the end. After take a screenshot of the
+# result, then, drop the table.
+
+USE loandb;
+
+CREATE TEMPORARY TABLE dsstudent.quantities
+	(table_name varchar(30),
+	row_quantity int);
+	
+# Is there a more succinct way of doing this without using several COUNT(*) queries?
+
+SELECT COUNT(*)
+FROM train; #307511
+
+SELECT COUNT(*)
+FROM bureau; #1716428
+
+SELECT COUNT(*)
+FROM bureau_balance; #27299925
+
+SELECT COUNT(*)
+FROM previous_application; #1670214
+
+SELECT COUNT(*)
+FROM installments_payments; #13605401
+
+SELECT COUNT(*)
+FROM POS_CASH_balance; #10001358
+
+SELECT COUNT(*)
+FROM credit_card_balance; #3840312
+
+INSERT INTO dsstudent.quantities (table_name, row_quantity)
+VALUES ('train', 307511),
+		('bureau', 1716428),
+		('bureau_balance', 27299925),
+		('previous_application', 1670214),
+		('installments_payments', 13605401),
+		('POS_CASH_balance', 10001358),
+		('credit_card_balance', 3840312);
+
+#2 Show the monthly and annual income
+
+SELECT AMT_INCOME_TOTAL AS annual_income, (AMT_INCOME_TOTAL/12) AS monthly_income
+FROM loandb.train;
+
+#3 Transform the “DAYS_BIRTH” column by dividing “-365” and round the value to the integer place. Call this column as “age.”
+
+SELECT ROUND(DAYS_BIRTH/-365) AS age
+FROM loandb.train;
+
+#4 Show the quantity of each occupation type and sort the quantity in descending order.
+SELECT OCCUPATION_TYPE AS occupation_type, COUNT(OCCUPATION_TYPE) AS quantity
+FROM loandb.train
+GROUP BY 1
+HAVING occupation_type IS NOT NULL;
+
+#5 In the field “DAYS_EMPLOYED”, the maximum value in this field is bad data, can you write a conditional logic to
+#mark these bad data as “bad data”, and other values are “normal data” in a new field called “Flag_for_bad_data”?
+
+# Is there a way to incorporate MAX into the CASE statement?

```

Comment:
You can use a SELECT-FROM statement in the condition:

`WHEN t.DAYS_EMPLOYED = (SELECT MAX(t1.DAYS_EMPLOYED) FROM loandb.train t1) THEN "bad data"`